lib/doorkeeper/orm/active_record/application.rb

```diff
@@ -11,14 +11,9 @@ class Application < ActiveRecord::Base
     end
     has_many :authorized_applications, through: :authorized_tokens, source: :application
 
-    def self.column_names_with_table
-      self.column_names.map { |c| "#{table_name}.#{c}" }
-    end
-
     def self.authorized_for(resource_owner)
-      joins(:authorized_applications).
-        where(AccessToken.table_name => { resource_owner_id: resource_owner.id, revoked_at: nil }).
-        group(column_names_with_table.join(','))
+      where(id: Doorkeeper::AccessToken.
+        where(resource_owner_id: resource_owner.id, revoked_at: nil))
     end
```

Comment:
There is no need in the JOIN and GROUP BY.  That method returns applications, that has access tokes for the requested resource_owner.

JOINS was joining on the access tokens table and GROUP was acting like DISTINCT. There is no need in such a complex and suspicious logic (especially with the `column_names` method).  You can see at the `spec/models/doorkeeper/applications_spec.rb` : 139